File: src/main/java/com/kmarzecki/communicator/service/BoardService.java
package com.kmarzecki.communicator.service;

import com.kmarzecki.communicator.api.board.BoardResponse;
import com.kmarzecki.communicator.api.board.CreateBoardRequest;
import com.kmarzecki.communicator.api.board.UpdateBoardRequest;

import java.security.Principal;
import java.util.List;
import java.util.UUID;

public interface BoardService {
    BoardResponse create(CreateBoardRequest request, Principal principal);
    List<BoardResponse> getBoards(Principal principal);
    void deleteBoard(UUID boardId, Principal principal);
    BoardResponse update(UpdateBoardRequest request, UUID id, Principal principal);
}

File: src/main/kotlin/com/kmarzecki/trollo/service/BoardService.kt
package com.kmarzecki.trollo.service

import com.kmarzecki.trollo.api.board.BoardResponse
import com.kmarzecki.trollo.api.board.CreateBoardRequest
import com.kmarzecki.trollo.api.board.LaneResponse
import com.kmarzecki.trollo.api.board.UpdateBoardRequest
import java.security.Principal
import java.util.*

interface BoardService {
    fun create(request: CreateBoardRequest, principal: Principal): BoardResponse
    fun getBoards(principal: Principal): List<BoardResponse>
    fun deleteBoard(boardId: UUID, principal: Principal)
    fun update(request: UpdateBoardRequest, id: UUID, principal: Principal): BoardResponse
    fun getLanesForBoard(boardId: UUID, principal: Principal): List<LaneResponse>
}